src/ccache.c

```diff
@@ -1040,7 +1040,7 @@ use_relative_paths_in_depfile(const char *depfile)
 		cc_log("Base dir not set, skip using relative paths");
 		return; // nothing to do
 	}
-	if (!has_absolute_include_headers) {
+	if (!has_absolute_include_headers && !conf->depend_mode) {
```

Comment:
Two thoughts:

1. Would you like to write a test case that fails without this fix?
2. I think that a better approach would be to update `has_absolute_include_headers` in `object_hash_from_depfile`, similar to what is done in `process_preprocessed_file`. What do you think?

Reply:
But the dependency file first has to be relativizated before ```object_hash_from_depfile``` is called since it only calculates the hash. 

Maybe this would be better? 

```
        if (conf->depend_mode)
        {
                has_absolute_include_headers = 1; // force the relativization of dependency paths since we do not yet know if the compiler produced absolute paths or not without parsing the dependency file first
        }

	if (generating_dependencies) {
		use_relative_paths_in_depfile(output_dep);
	}

	if (conf->depend_mode) {
		struct file_hash *object_hash =
			object_hash_from_depfile(output_dep, depend_mode_hash);
```

Reply:
I've separated this issue into #366